src/app/wishlists/wishlist-fetch.ts

```diff
@@ -16,45 +14,47 @@ export function fetchWishList(): ThunkResult<Promise<void>> {
       return;
     }
 
-    fetch(wishListSource)
-      .then((result) => result.text())
-      .then((resultText) => transformAndStoreWishList(resultText, 'Fetch Wish List', dispatch));
+    const wishListResponse = await fetch(wishListSource);
+    const wishListText = await wishListResponse.text();
+
+    dispatch(transformAndStoreWishList(wishListText, 'Fetch Wish List'));
   };
 }
 
 export function transformAndStoreWishList(
   wishListResult: string,
-  eventName: string,
-  dispatch?: ThunkDispatch<RootState, {}, AnyAction>
-) {
-  const wishListAndInfo = toWishList(wishListResult);
-  ga('send', 'event', 'Rating Options', eventName);
-
-  if (wishListAndInfo.wishListRolls.length > 0) {
-    if (dispatch) {
-      dispatch(loadWishLists(wishListAndInfo));
+  eventName: string
+): ThunkResult<Promise<void>> {
+  return async (dispatch) => {
+    const wishListAndInfo = toWishList(wishListResult);
+    ga('send', 'event', 'Rating Options', eventName);
+
+    if (wishListAndInfo.wishListRolls.length > 0) {
+      if (dispatch) {
```

Comment:
`dispatch` is now always available, so you don't need to check it and switch to `store.dispatch`.

Reply:
_oof_ but easy fix.